examples/prostate/pt/trainers/supervised_trainer.py

```diff
@@ -0,0 +1,353 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+import json
+import os
+
+import numpy as np
+import torch
+from torch.utils.tensorboard import SummaryWriter
+
+from nvflare.apis.dxo import DXO, DataKind, MetaKey, from_shareable
+from nvflare.apis.event_type import EventType
+from nvflare.apis.executor import Executor
+from nvflare.apis.fl_constant import FLContextKey, ReturnCode
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.shareable import Shareable, make_reply
+from nvflare.apis.signal import Signal
+from nvflare.app_common.app_constant import AppConstants
+from nvflare.app_common.pt.pt_fedproxloss import PTFedProxLoss
+
+
+class SupervisedTrainer(Executor):
+    def __init__(
+        self,
+        train_config_filename,
+        aggregation_epochs: int = 1,
+        train_task_name: str = AppConstants.TASK_TRAIN,
+        submit_model_task_name=AppConstants.TASK_SUBMIT_MODEL,
+    ):
+        """Simple Supervised Trainer.
+
+        Args:
+            train_config_filename: directory of config file.
+            aggregation_epochs: the number of training epochs for a round.
+                This parameter only works when `aggregation_iters` is 0. Defaults to 1.
+            train_task_name: name of the task to train the model.
+            submit_model_task_name: name of the task to submit the best local model.
+
+        Returns:
+            a Shareable with the updated local model after running `execute()`
+            or the best local model depending on the specified task.
+        """
+        super().__init__()
+        # trainer init happens at the very beginning, only the basic info regarding the trainer is set here
+        # the actual run has not started at this point
+        self.train_config_filename = train_config_filename
+        self.aggregation_epochs = aggregation_epochs
+        self.train_task_name = train_task_name
+        self.submit_model_task_name = submit_model_task_name
+        self.best_acc = 0.0
+
+        # Epoch counter
+        self.epoch_of_start_time = 0
+        self.epoch_global = 0
+
+        # FedProx related
+        self.fedproxloss_mu = 0.0
+        self.criterion_prox = None
+
+    def _initialize_trainer(self, fl_ctx: FLContext):
+        # when the run starts, this is where the actual settings get initialized for trainer
+
+        # Set the paths according to fl_ctx
+        self.app_root = fl_ctx.get_prop(FLContextKey.APP_ROOT)
```

Comment:
Please use Workspace API to manage config loading, instead of use app_root:

engine = fl_ctx.get_engine()
ws = engine.get_workspace()
app_config_dir = ws.get_app_config_dir(fl_ctx.get_run_number())

Then the config file name is:
config_file_full_path = os.path.join(app_config_dir, config_file_name)

Here config_file_name is simply "config_train.json".